rts/Sim/Misc/Team.cpp

```diff
@@ -221,10 +221,21 @@ void CTeam::GiveEverythingTo(const unsigned toTeam)
 
 	const auto& teamUnits = unitHandler.GetUnitsByTeam(teamNum);
 
+	// Optimistically give *all* of team's unit limit to target
+	// If some transfers fail, need to partially revert this
+	// to ensure this.maxUnits == this.numUnits
+	target->maxUnits += maxUnits;
+	maxUnits = 0;
+
 	// NB: can not be a ranged loop since ChangeTeam removes [i] from teamUnits on success
 	for (size_t i = 0; i < teamUnits.size(); ) {
 		i += (!teamUnits[i]->ChangeTeam(toTeam, CUnit::ChangeGiven));
 	}
+
```

Comment:
I was about to say that `numUnits` below should probably be `teamUnits.size()` but I think they are equivalent. Consider an extra safety check
```suggestion
	assert(numUnits == teamUnits.size());
```